src/pyproject_hooks/_impl.py

```diff
@@ -39,19 +45,35 @@ def __init__(self, backend_name, backend_path, message):
 class HookMissing(Exception):
     """Will be raised on missing hooks (if a fallback can't be used)."""
 
-    def __init__(self, hook_name):
+    def __init__(self, hook_name: str) -> None:
         super().__init__(hook_name)
         self.hook_name = hook_name
 
 
 class UnsupportedOperation(Exception):
     """May be raised by build_sdist if the backend indicates that it can't."""
 
-    def __init__(self, traceback):
+    def __init__(self, traceback: str) -> None:
         self.traceback = traceback
 
 
-def default_subprocess_runner(cmd, cwd=None, extra_environ=None):
+class SubprocessRunner(Protocol):
+    """A protocol for the subprocess runner."""
+
+    def __call__(
+        self,
+        cmd: List[str],
+        cwd: Optional[str] = None,
+        extra_environ: Optional[Dict[str, str]] = None,
```

Comment:
It's considered [good practice](https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#wide-vs-narrow-types) to use collection ABCs for function parameters:

```suggestion
        cmd: Sequence[str],
        cwd: Optional[str] = None,
        extra_environ: Optional[Mapping[str, str]] = None,
```

Reply:
Done.